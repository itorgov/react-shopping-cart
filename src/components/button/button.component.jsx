import React from 'react';
import classNames from "classnames";

import styles from './button.module.scss';

export const Button = ({variant, children: content}) => (
    <button className={classNames([
        styles.button,
        styles[variant || 'primary'],
    ])}
    >
        {content}
    </button>
);
