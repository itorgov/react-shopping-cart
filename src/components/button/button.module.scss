.button {
  font-size: 1.4rem;
  letter-spacing: 3px;
  text-transform: uppercase;
  background: none;
  border: 0;
  border-radius: 10rem;
  padding: 1.5rem 3rem;
  transition: transform 0.2s, box-shadow 0.2s;

  display: flex;
  justify-content: center;
  align-items: center;

  & svg {
    height: 2.5rem;
    width: 2.5rem;
    margin-left: 1rem;
    fill: currentColor;
  }

  &:hover {
    transform: translateY(-2px);
    box-shadow: 0 1rem 2rem rgba(var(--color-secondary-rgb), 0.2);
  }

  &:active {
    transform: translateY(-1px);
    box-shadow: 0 0.5rem 1rem rgba(var(--color-secondary-rgb), 0.2);
  }

  &:focus {
    outline: none;
  }
}

.black {
  background-color: var(--color-text-dark);
  color: var(--color-white);
}
