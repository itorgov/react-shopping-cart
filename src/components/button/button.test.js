import React from "react";
import {shallow} from "enzyme";

import {Button} from './button.component';

describe('<Button />', () => {
    it('expect to render', () => {
        expect(shallow(<Button />)).toMatchSnapshot();
    });

    it('renders content inside', () => {
        const wrapper = shallow(<Button>Test string</Button>);
        expect(wrapper.text()).toEqual('Test string');
    });
});
