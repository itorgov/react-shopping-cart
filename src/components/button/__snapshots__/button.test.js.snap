// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> expect to render 1`] = `
<button
  className="button primary"
/>
`;
